### DISCUSSION TITLE
I am unable to share drizzle schema between two backend

### DISCUSSION THREAD

**Discussion by riteshreg** [t3-oss/create-t3-turbo#1249] [Q&A] [answered]
I tried to import the schema from the @acme/db package, but I was unable to compile it. I have fixed some issues, but I am still unable to resolve this particular problem. the error is 
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts"
![Screenshot (120)](https://github.com/user-attachments/assets/63eb3f97-0c9a-493d-986f-ca9fb798c438)

my [nestjs ](https://www.nestjs.com)tsconfig.json
{
  "extends": "@acme/tsconfig/base.json",
  "compilerOptions": {
    "incremental": false,
    "moduleResolution": "node16",
    "module": "Node16",
    "target": "ES2021",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "noEmit": false,
    "outDir": "./dist",
    "baseUrl": "./",
    "paths": {
      "@acme/*": ["../../packages/*/dist"]
    }
  },
  "exclude": ["node_modules", "dist", "build"]
}

Any assistance would be greatly appreciated!

**Comment by CarlosZiegler** [comment]
"@acme/": ["../../packages//dist"] => this will build only js files and you trying to import ts from that or ?

**Reply by riteshreg** [reply]
yes I am trying to import ts file.  I tried to convert the nestjs commonjs to module but it didn't work. I am not able to figure out the solution.

**Comment by riteshreg** [comment, marked as answer]
Here is my solution,
I am using nestjs as my second backend

1. In packages/db update the tscofig so that javascript file also emit in dist (my whole tsconfig.json)

{
  "extends": "@acme/tsconfig/internal-package.json",
  "include": ["src"],
  "exclude": ["node_modules", "dist"],
  "compilerOptions": {
    "module": "esnext", // Use modern module system
    "declaration": true,
    "declarationMap": true,
    "emitDeclarationOnly": false, // Ensure this is false to emit JavaScript
    "noEmit": false,
    "outDir": "./dist" // Specify output directory
  }
}


2. In package.json  export .js file also (my package.json exports)
 "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./client": {
      "types": "./dist/client.d.ts",
      "default": "./dist/client.js"
    },
    "./schema": {
      "types": "./dist/schema.d.ts",
      "default": "./dist/schema.js"
    }
  },
  

3. As I mention I am using nestjs as my second backend so I converted it to module type (here is my nestjs tsconfig)
{
  "compilerOptions": {
    "esModuleInterop": true,
    "moduleResolution": "nodenext",
    "module": "NodeNext",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false
  }
}
